share/efetch.sh

```diff
@@ -105,7 +105,12 @@ __efetch_load_info()
         destination="${TMPDIR:-/tmp}"
     else
         destination=${urls[-1]}
-        unset urls[-1]
+        
+        if [[ ${BASH_VERSION} == 4.2* ]]; then
+            unset urls[${#urls[@]}-1]
```

Comment:
The above fails on bash 4.2

Reply:
This should be futureproofed like this:
`   if [[ ${BASH_VERSINFO[0]} -gt 4 || ( ${BASH_VERSINFO[0] -eq 4 && ${BASH_VERSINFO[1] -gt 1 ) ]] ; then`

Reply:
well... unless it _only_ fails on 4.2 and works on 4.1 and 4.3

Reply:
It fails on anything <= 4.2. So yeah, I should make this check better.

Reply:
I fixed the above check to always use the safer backwards compatible version instead of the newer one.